From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Tue, 4 Feb 2014 17:42:03 +0100
Subject: [PATCH] adjust PackageInfo for zip files

see commit 7fe9808f009af2f0a4da58663eacd2caccdcaafe for reference

Reviewed-On: https://gerrit.libreoffice.org/7975/
Change-Id: I060021150f8ef976ffeb417b18dda31367e549b8

---
 solenv/gbuild/extensions/post_PackageInfo.mk | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/solenv/gbuild/extensions/post_PackageInfo.mk b/solenv/gbuild/extensions/post_PackageInfo.mk
index 03a0f24..0c3429f 100644
--- a/solenv/gbuild/extensions/post_PackageInfo.mk
+++ b/solenv/gbuild/extensions/post_PackageInfo.mk
@@ -56,8 +56,8 @@ define gb_PackageInfo_emit_l10n_for_one_ressource
 
 endef
 
-define gb_PackageInfo_emit_l10n_for_one_uifile
-@echo "$(gb_UIConfig_INSTDIR)/$(2)/ui/res/$(1)/$(3)" >> $(gb_PackageInfo_get_target)/l10n-$(1).files
+define gb_PackageInfo_emit_l10n_for_one_uizip
+@echo "$(gb_UIConfig_INSTDIR)/$(2)/ui/res/$(1).zip" >> $(gb_PackageInfo_get_target)/l10n-$(1).files
 
 endef
 
@@ -70,7 +70,9 @@ define gb_PackageInfo_emit_l10n_for_one_lang
 @touch $(foreach suf,executables libraries files,$(gb_PackageInfo_get_target)/l10n-$(1).$(suf))
 $(if $(filter-out qtz en-US,$(1)),$(foreach packagedir,$(patsubst %/,%,$(gb_AllLangPackage_ALLDIRS)),$(call gb_PackageInfo_emit_l10n_for_one_alllangpackage,$(packagedir),$(1))))
 $(if $(filter $(gb_AllLangResTarget_LANGS),$(1)),$(foreach target,$(gb_AllLangResTarget_ALLTARGETS),$(call gb_PackageInfo_emit_l10n_for_one_ressource,$(target),$(1))))
-$(foreach uifile,$(gb_UIConfig_ALLFILES),$(call gb_PackageInfo_emit_l10n_for_one_uifile,$(1),$(firstword $(subst :,$(WHITESPACE),$(uifile))),$(lastword $(subst :,$(WHITESPACE),$(uifile)))))
+$(foreach uizip,\
+    $(sort $(foreach uifile,$(gb_UIConfig_ALLFILES),$(firstword $(subst :,$(WHITESPACE),$(uifile))))),\
+    $(call gb_PackageInfo_emit_l10n_for_one_uizip,$(1),$(uizip)))
 $(if $(filter $(gb_Configuration_LANGS),$(1)),$(foreach configfile,Langpack- res/fcfg_langpack_ res/registry_,$(call gb_PackageInfo_emit_l10n_for_one_configfile,$(1),$(configfile))))
 @echo "$(foreach suf,executables libraries files,$(gb_PackageInfo_get_target)/l10n-$(1).$(suf)) \\" >> $(WORKDIR)/Dep/packageinfo.d
 
-- 
1.8.3.2

